### PR TITLE
NIAD-1504 - Separation of large ehrExtract and normal attachments

### PR DESCRIPTION
**Pull request by ElliotB-96**
Distinguished between documents and ehrExtracts being treated as an attachment

### CODE REVIEW

**Review by bbell-kainos — APPROVED** [review]

File: e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/AwaitHelper.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 public class AwaitHelper {
 
-    protected static final int WAIT_FOR_IN_SECONDS = 60;
+    protected static final int WAIT_FOR_IN_SECONDS = 90;
```
Comment (sylwestergryzio) [inline e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/AwaitHelper.java:13]: I think this is for slower machines? ;) At some point we would need to stop raising that value. for now ok.

File: e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java [hunk 2]
```diff
@@ -217,10 +231,45 @@ private void assertHappyPathWithDocs(String conversationId, String fromODSCode,
...
+        assertThat(gpcAccessStructured.get("taskId")).isNotNull();
+        assertThat(gpcAccessStructured.get("messageId")).isNotNull();
+        assertThat(gpcAccessStructured.get("sentToMhs")).isNotNull();
+    }
+
+    private Document fetchSentToMhsForDocuments(String conversationId) {
+        var gpcAccessDocument = (Document) Mongo.findEhrExtractStatus(conversationId).get(GPC_ACCESS_DOCUMENT);
+        var documentList = gpcAccessDocument.get("documents", Collections.emptyList());
+        if (documentList == null || documentList.isEmpty())
+            return null;
```
Comment (sylwestergryzio) [inline e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java:257]: this null will throw null pointer exception in the test right? e.g. assertThat(sentToMhs.get("messageId")).isNotNull(); would throw exception? if that is the case maybe null is some king of exception that would have to be thrown?

File: e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java [hunk 3]
```diff
@@ -217,10 +231,45 @@ private void assertHappyPathWithDocs(String conversationId, String fromODSCode,
...
+
+        var firstDocSentToMhs = (Document) documentList.get(0);
+
+        return (Document) firstDocSentToMhs.get("sentToMhs");
+    }
+
+    private Document accessStructuredWithAttachmentThatHasBeenSent(String conversationId) {
+        var gpcAccessStructured = (Document) Mongo.findEhrExtractStatus(conversationId).get(GPC_ACCESS_STRUCTURED);
+        if (gpcAccessStructured == null)
+            return null;
```
Comment (sylwestergryzio) [inline e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java:267]: same question here. if it happens to return null then in test we would call null.get("...")?

File: e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java [hunk 4]
```diff
@@ -217,10 +231,45 @@ private void assertHappyPathWithDocs(String conversationId, String fromODSCode,
...
+        var attachment = (Document) gpcAccessStructured.get("attachment");
+        if (attachment != null && attachment.get("sentToMhs") != null) {
+            return attachment;
+        }
+        return null;
```
Comment (sylwestergryzio) [inline e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java:272]: same here. think how to limit number of return statements.

File: e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java [hunk 5]
```diff
@@ -236,16 +285,13 @@ private Document theDocumentTaskUpdatesTheRecord(String conversationId) {
         return getFirstDocumentIfItHasObjectNameOrElseNull(gpcAccessDocument);
     }
 
-    private Document emptyDocumentTaskIsCreated(String conversationId) {
+    private List<Object> emptyDocumentTaskIsCreated(String conversationId) {
```
Comment (sylwestergryzio) [inline e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java:288]: Why Object? are there different items that documents being returned?

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/request/EhrExtractRequestHandler.java [hunk 6]
```diff
@@ -126,14 +126,31 @@ public void handleContinue(String conversationId, String payload) {
             ehrExtractStatusService.updateEhrExtractStatusContinue(conversationId)
                 .ifPresent(ehrExtractStatus -> {
                     var documents = ehrExtractStatus.getGpcAccessDocument().getDocuments();
+                    if (ehrExtractStatus.getGpcAccessStructured() != null) {
```
Comment (sylwestergryzio) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/request/EhrExtractRequestHandler.java:129]: I would put that into seperate method for readability - i.e. looks like this methods logs something when a codition is met

**Review by sylwestergryzio — CHANGES_REQUESTED** [review]

File: e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/AwaitHelper.java (hunk 1 above)
Comment (ElliotB-96) [reply e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/AwaitHelper.java:13]: I've reduced this again, will review it during the next pipeline run

File: e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java (hunk 5 above)
Comment (ElliotB-96) [reply e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java:288]: it's a method that isn't being used, I have removed it.

**Review by bbell-kainos — APPROVED** [review]

**Review by sylwestergryzio — APPROVED** [review]

**Pull request merged**